### PR TITLE
ARISTOTLE: handle dip and strike depending from the rupture type; handle time

### PR DESCRIPTION
**Pull request by ptormene**
- If a rupture model xml is uploaded, dip and strike are read from it and the corresponding form fields are disabled.
- If the rupture or finite fault is obtained from the USGS:
    - If the rupture is of type "Point", dip an strike form fields are editable and they initially have the default values
    - Otherwise, dip and strike form fields are disabled and blank, so they are passed as None when setting the parameters for the calculation

I am also retrieving the timestamp of the event, converted from UTC to the local time (given lon and lat). I added a non-editable field displaying that timestamp, plus another pre-populated field to set the `time_event` parameter to pass as a parameter of the calculation.
I added a local_timestamp validator and I am visualizing in the outputs page the local timestamp and how much time after the event the job was started (see screenshot):
image

NOTE: converting from UTC to the local time given lon and lat requires the python package timezonefinder, that is not in the engine requirements and that I am importing at runtime, handling the error in case it is not installed. It should not be a problem adding this dependency just for ARISTOTLE installations.

Companion of https://gitlab.openquake.org/hazard/mosaic/mosaic/-/merge_requests/71

### CODE REVIEW

File: openquake/server/views.py [hunk 1]
```diff
@@ -849,31 +864,40 @@ def aristotle_run(request):
...
+        maximum_distance_stations, station_data_file
     """
     res = aristotle_validate(request)
     if isinstance(res, HttpResponse):  # error
         return res
-    (rupdic, maximum_distance, trt,
+    (rupdic, local_timestamp, time_event, maximum_distance, trt,
      truncation_level, number_of_ground_motion_fields,
      asset_hazard_distance, ses_seed, maximum_distance_stations,
      station_data_file) = res
```
Comment (micheles) [inline openquake/server/views.py:880]: Having so many positional parameters is not good. We will have to think of something (in another PR).

File: openquake/hazardlib/shakemap/parsers.py [hunk 2]
```diff
@@ -200,6 +202,38 @@ def get_array_usgs_xml(kind, grid_url, uncertainty_url=None):
             'USGS xml grid file could not be found at %s' % grid_url) from e
 
 
+def utc_to_local_time(utc_timestamp, lon, lat):
+    try:
+        # NOTE: optional dependency needed for ARISTOTLE
+        from timezonefinder import TimezoneFinder
```
Comment (micheles) [inline openquake/hazardlib/shakemap/parsers.py:208]: Let's move the import check at Django startup time

**Comment by micheles** [comment]
LGTM apart from the comment about the ImportError

File: openquake/server/apps.py [hunk 3]
```diff
@@ -38,6 +38,16 @@ def ready(self):
...
+        if settings.APPLICATION_MODE == 'ARISTOTLE':
+            try:
+                # NOTE: optional dependency needed for ARISTOTLE
+                from timezonefinder import TimezoneFinder  # noqa
+            except ImportError:
+                raise ImportError(
+                    'The python package "timezonefinder" is not installed.'
+                    ' It is required in order to convert the UTC time to'
+                    ' the local time of the event. You can install it'
+                    ' running: pip install timezonefinder==6.5.2')
```
Comment (ptormene) [inline openquake/server/apps.py:50]: I added the check here, raising the error if the webui is launched in ARISTOTLE mode and timezonefinder is not installed. I suppose the check in parsers.py can still remain, so the error will be raised also when running that code independently from the webui.

**Pull request merged**